### PR TITLE
Add build configuration to generate runtime packs for WebAssembly.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
Based on the similar ios and Android PRs.

**Comment by marek-safar** [comment]
I think we should replace `WebAssembly` with `Browser` here to align with #33328

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -397,6 +397,7 @@ jobs:
     - tvOS_arm64
     # - iOS_arm # https://github.com/dotnet/runtime/issues/34465
     - iOS_x64
+    - WebAssembly_wasm
```
Comment (akoeplinger) [inline eng/pipelines/runtime.yml:400]: this doesn't need to be here, it's already in the build template below (the purpose of this is to exercise a Debug/Release build mix on PRs: https://github.com/dotnet/runtime/pull/34835)

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -508,6 +509,7 @@ jobs:
...
```
Comment (akoeplinger) [inline eng/pipelines/runtime.yml:512]: same here, you can remove it from this one.

**Review by akoeplinger — COMMENTED** [review]
I agree with Marek about the WebAssembly vs. Browser as the TargetOS, but not sure if we need to block this PR on that.

**Comment by marek-safar** [comment]
Let's land this with right names (makes no sense to rename it in few days)

**Comment by vargaz** [comment]
Afaik browser is used int the API, it has little to do with internal architecture/os names. We can split these when we have multiple versions of wasm for server etc, right now we don't.

**Review by safern — APPROVED** [review]

**Comment by safern** [comment]
cc: @dagood @NikolaMilosavljevic

**Comment by marek-safar** [comment]
@vargaz could you please update the PR to align with #34940 outcome

**Comment by vargaz** [comment]
What needs to be done here ? Change the os to Browser everywhere ?

**Comment by marek-safar** [comment]
Yes + resolving the conflicts

**Comment by vargaz** [comment]
Rebased, please re-review.

**Review by marek-safar — APPROVED** [review]

**Pull request merged**